packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx

```diff
@@ -63,31 +75,176 @@ const useStyles = makeStyles(({ palette, spacing, typography }) => ({
+        <IconVerifiedGreen style={styles.checkIcon} />
+      </View>
+      {renderDescription()}
+    </View>
+  )
+}
+
+const DetailsTileOwnerSection = ({
+  renderDescription,
+  nftCollection,
+  followee,
+  tippedUser
+}: HasAccessProps) => {
+  const styles = useStyles()
+  const neutral = useColor('neutral')
+
+  return (
+    <View style={styles.root}>
+      <View style={styles.titleContainer}>
+        {nftCollection && (
+          <IconCollectible fill={neutral} width={16} height={16} />
+        )}
+        {(followee || tippedUser) && (
+          <IconSpecialAccess fill={neutral} width={16} height={16} />
+        )}
+        <Text style={styles.title}>
+          {nftCollection ? messages.collectibleGated : messages.specialAccess}
+        </Text>
+      </View>
+      {renderDescription()}
+    </View>
+  )
+}
+
+type DetailsTilePremiumAccessProps = {
+  premiumConditions: PremiumConditions
+  isOwner: boolean
+}
+
+export const DetailsTileHasAccess = ({
+  premiumConditions,
+  isOwner
+}: DetailsTilePremiumAccessProps) => {
+  const styles = useStyles()
+
+  const { nftCollection, followee, tippedUser } =
+    useSpecialAccessEntity(premiumConditions)
+
+  const collectionLink = useMemo(() => {
+    if (!nftCollection) return ''
+
+    const { chain, address, externalLink } = nftCollection
+    if (chain === Chain.Eth && 'slug' in nftCollection!) {
+      return `https://opensea.io/collection/${nftCollection.slug}`
+    } else if (chain === Chain.Sol) {
+      const explorerUrl = `https://explorer.solana.com/address/${address}`
+      return externalLink ? new URL(externalLink).hostname : explorerUrl
+    }
+
+    return ''
+  }, [nftCollection])
+
+  const { onPress: handlePressCollection } = useLink(collectionLink)
+
+  const renderOwnerDescription = useCallback(() => {
+    if (nftCollection) {
       return (
-        <View>
-          <Text style={styles.description}>yolo</Text>
+        <View style={styles.descriptionContainer}>
+          <Text>
+            <Text style={styles.description}>
+              {messages.ownerCollectibleGatedPrefix}
+            </Text>
+            <Text
+              onPress={handlePressCollection}
+              style={[styles.description, styles.collectionName]}
+            >
+              {nftCollection.name}
+            </Text>
+          </Text>
         </View>
       )
     }
     if (followee) {
       return (
-        <View>
-          <Text style={styles.description}>yolo</Text>
+        <View style={styles.descriptionContainer}>
+          <Text>
+            <Text style={styles.description}>{messages.ownerFollowGated}</Text>
+          </Text>
+        </View>
+      )
+    }
+    if (tippedUser) {
+      return (
+        <View style={styles.descriptionContainer}>
+          <Text>
+            <Text style={styles.description}>{messages.ownerTipGated}</Text>
+          </Text>
+        </View>
+      )
+    }
+
+    // should not reach here
+    return null
```

Comment:
Is there a better way to represent this? Maybe at least a `console.error()`?

Reply:
yeah let me do a console.warn here thx

Reply:
could model this with a discriminated union and then the compiler would enforce that it's impossible for this to happen :)